p2p/client.go

```diff
@@ -358,7 +357,22 @@ func validateSignatureResp(
 
 	err := identity.ValidateDIDBytes(resp.Signature.SignerId, receiver)
 	if err != nil {
-		return centerrors.New(code.AuthenticationFailed, err.Error())
+		return centerrors.New(code.AuthenticationFailed, fmt.Sprintf("signature invalid with err: %s", err.Error()))
+	}
+
+	tm, err := model.Timestamp()
+	if err != nil {
+		return centerrors.New(code.AuthenticationFailed, fmt.Sprintf("cannot get model timestamp : %s", err.Error()))
+	}
+	signingRoot, err := model.CalculateSigningRoot()
+	if err != nil {
+		return centerrors.New(code.AuthenticationFailed, fmt.Sprintf("failed to calculate signing root: %s", err.Error()))
+	}
+
+	err = s.idService.ValidateSignature(receiver, resp.Signature.PublicKey, resp.Signature.Signature, signingRoot, tm)
+
+	if err != nil {
+		return centerrors.New(code.AuthenticationFailed, fmt.Sprintf("signature invalid with err: %s", err.Error()))
```

Comment:
`centerrors` are deprecated, please use `errors` instead

Reply:
what about  package code?

Reply:
Ignore this